core/src/main/java/com/amplifyframework/datastore/DataStorePlugin.java

```diff
@@ -39,4 +43,17 @@ public final CategoryType getCategoryType() {
     @Override
     public void initialize(@NonNull Context context) throws AmplifyException {
     }
+
+    /**
+     * Query the DataStore to find all items of the requested model (by name).
+     * NOTE: Private method and should not be part of {@link DataStoreCategory}
+     * @param modelName name of the Model to query
+     * @param onQueryResults Called when a query successfully returns 0 or more results
+     * @param onQueryFailure Called when a failure interrupts successful completion of a query
+     */
+    public void query(
```

Comment:
The reason we need the `String`-arg is to specify which model we are querying because Class<? extends Model> cannot distinguish between different models (There are no `Model` classes when calling from Flutter, just one `Model` class called `SerializedModel` that encapsulates all the models). 

Between 1 and 2, option 2 was not picked in the design review as the group explicitly didn't want to expose this as a functionality to native library customers. I'm open for suggestions for option 1 however not clear on the advantages.

Reply:
This isn't going to work for reasons that are clear to me now - we are talking as I write this and we will post some additional design notes! :-).

Reply:
^ Clarified offline: `Class<T>` doesn't exist from Flutter, since we aren't going to code-gen the Java models.